module/zfs/dsl_dir.c

```diff
@@ -2282,7 +2293,104 @@ dsl_dir_remove_livelist(dsl_dir_t *dd, dmu_tx_t *tx, boolean_t total)
 	}
 }
 
+static int
+dsl_dir_activity_in_progress(dsl_dir_t *dd, dsl_dataset_t *ds,
+    zfs_wait_activity_t activity, boolean_t *in_progress)
+{
+	int error = 0;
+
+	ASSERT(MUTEX_HELD(&dd->dd_activity_lock));
+
+	switch (activity) {
+	case ZFS_WAIT_DELETEQ: {
+#ifdef _KERNEL
+		objset_t *os;
+		error = dmu_objset_from_ds(ds, &os);
+		if (error != 0)
+			break;
+
+		if (dmu_objset_type(os) != DMU_OST_ZFS ||
+		    zfs_get_vfs_flag_unmounted(os)) {
```

Comment:
I think zfs_get_vfs_flag_unmounted() returns FALSE if this is not mounted.  It only returns TRUE if the zfsvfs_t exists but z_unmounted is set.  Is that the semantic we want?  I think we want to also return FALSE if the fs is not mounted.

Reply:
Do we have a concise way to check if it's mounted?